src/js/controller.js

```diff
@@ -0,0 +1,185 @@
+import { validateTodo } from "./filterValidationOnTodo.js";
+import { todoActions, urgency, category } from "./consts.js";
+
+export class Controller {
+  constructor(view, model) {
+    this.model = model;
+    this.view = view;
+
+    this.view.updateHeaderDate();
+    this.view.bindUndo(this.handleUndo);
+    this.view.bindRedo(this.handleRedo);
+    this.view.bindAddTodo(this.handleAddTodo);
+    this.view.bindFilterUpdate(this.handleFilterUpdate);
+    this.view.bindCheckBoxUpdate(this.handleCheckBoxUpdate);
+    this.view.bindSearchBoxUpdate(this.handleSearchBoxUpdate);
+    this.view.bindClearSearchBtn();
+    this.view.bindToggleCompletionOfSelection(this.toggleCompletionOfSelection);
+    this.view.bindClearSelection(this.clearSelection);
+    this.view.bindDeleteSelectedTodos(this.deleteSelectedTodos);
+
+    this.model.bindStateChangeHandler(this.render);
+
+    this.callbacksForTodo = {
+      handleTodoToggle: this.handleTodoToggle,
+      handleTogglingFromSelected: this.handleTogglingFromSelected,
+      handleDeleteTodo: this.handleDeleteTodo,
+      handleEditTodo: this.handleEditTodo,
+    };
+
+    this.render(
+      this.model.allTodos,
+      this.model.currentlySelected,
+      this.model.filterData
+    );
+  }
+
+  render = (todos, currentlySelectedIds, filterData) => {
+    const filteredTodos = todos.filter((todo) =>
+      validateTodo(todo, filterData)
+    );
+    // console.log(filteredTodos);
+    this.view.analyticsUpdater.resetCounts();
+    this.view.displayTodos(
+      filteredTodos,
+      currentlySelectedIds,
+      this.callbacksForTodo
+    );
+    filteredTodos.forEach((todo) =>
+      this.view.analyticsUpdater.updateCounts(todo)
+    );
+    this.view.analyticsUpdater.updateAnalyticsOnView();
+  };
+
+  createTodoObject = ({ counter, title, urgency, category }) => {
```

Comment:
defaults should be handled as
```js
createTodoObject = ({counter, title='', urgency=0, category=0}) => {
```

Also, use `URGENCY.LOW` / `CATEGORY.LOW` as the default and not 0

Reply:
handled defaults in the function called "handleAddTodo" where i am getting inputs from the view.